lasagne/layers/pool.py

```diff
@@ -291,6 +292,51 @@ def __init__(self, incoming, pool_size, stride=None,
 # TODO: add MaxPool3DLayer
 
 
+class Upscale2DLayer (Layer):
+    """
+    2D upscaling layer layer
+
+    Performs 2D upscaling over the two trailing axes of a 4D input tensor.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or tuple
+        The layer feeding into this layer, or the expected input shape.
+
+    scale_factor : integer or iterable
+        The scale factor in each dimensions. If an integer, it is promoted to
+        a square scale factor region. If an iterable, it should have two
+        elements.
+
+    **kwargs
+        Any additional keyword arguments are passed to the :class:`Layer`
+        superclass.
+    """
+
+    def __init__(self, incoming, scale_factor, **kwargs):
+        super(Upscale2DLayer, self).__init__(incoming, **kwargs)
+
+        self.scale_factor = as_tuple(scale_factor, 2)
+
+    def get_output_shape_for(self, input_shape):
+        output_shape = list(input_shape)  # copy / convert to mutable list
+        output_shape[2] = input_shape[2] * self.scale_factor[0] \
+            if input_shape[2] is not None else None
+        output_shape[3] = input_shape[3] * self.scale_factor[1] \
+            if input_shape[3] is not None else None
```

Comment:
If `input_shape[2] is None`, then `output_shape[2] is None` already, no need to set it again. So you can simplify it to:
```
output_shape = list(input_shape)
if output_shape[2] is not None:
    output_shape[2] *= self.scale_factor[0]
if output_shape[3] is not None:
    output_shape[3] *= self.scale_factor[1]
return tuple(output_shape)
```